public/app/features/variables/pickers/OptionsPicker/reducer.ts

```diff
@@ -78,6 +78,10 @@ const updateOptions = (state: OptionsPickerState): OptionsPickerState => {
 
     return { ...option, selected };
   });
+  if (Array.isArray(state.options) && state.options.length > OPTIONS_LIMIT) {
+    const options = state.options;
+    state.options = options.slice(0, OPTIONS_LIMIT);
+  }
```

Comment:
Thank you for contributing @guoqn this looks good but I think that you could change this to call `applyLimit` function instead and remove other calls to applyLimit as `updateOptions` is called on all the reducer calls.

Thank you
Also, we need some tests for this new logic. 
```suggestion
  state.options = applyLimit(state.options);
```

Reply:
Wow, this looks a little bit more neat

Reply:
but, the other two USES of 'applyLimit' have their own logic and are not recommended